src/state.rs

```diff
@@ -0,0 +1,372 @@
+                }
+                BevyWindowEvent::MouseWheel(e) => {
+                    world.send_event(e);
+                }
+                BevyWindowEvent::PinchGesture(e) => {
+                    world.send_event(e);
+                }
+                BevyWindowEvent::RotationGesture(e) => {
+                    world.send_event(e);
+                }
+                BevyWindowEvent::DoubleTapGesture(e) => {
+                    world.send_event(e);
+                }
+                BevyWindowEvent::PanGesture(e) => {
+                    world.send_event(e);
+                }
+                BevyWindowEvent::TouchInput(e) => {
+                    world.send_event(e);
+                }
+                BevyWindowEvent::KeyboardInput(e) => {
+                    world.send_event(e);
+                }
+                BevyWindowEvent::KeyboardFocusLost(e) => {
+                    world.send_event(e);
+                }
+            }
+        }
+
+        if !buffered_events.is_empty() {
+            world
+                .resource_mut::<Events<BevyWindowEvent>>()
+                .send_batch(buffered_events);
+        }
+    }
+}
+
+impl CompositorHandler for SmithayRunnerState {
+    fn scale_factor_changed(
+        &mut self,
+        _conn: &Connection,
+        _qh: &smithay_client_toolkit::reexports::client::QueueHandle<Self>,
+        _surface: &smithay_client_toolkit::reexports::client::protocol::wl_surface::WlSurface,
+        _new_factor: i32,
+    ) {
+    }
+
+    fn transform_changed(
+        &mut self,
+        _conn: &Connection,
+        _qh: &smithay_client_toolkit::reexports::client::QueueHandle<Self>,
+        _surface: &smithay_client_toolkit::reexports::client::protocol::wl_surface::WlSurface,
+        _new_transform: smithay_client_toolkit::reexports::client::protocol::wl_output::Transform,
+    ) {
+    }
+
+    fn frame(
+        &mut self,
+        _conn: &Connection,
+        _qh: &smithay_client_toolkit::reexports::client::QueueHandle<Self>,
+        _surface: &smithay_client_toolkit::reexports::client::protocol::wl_surface::WlSurface,
+        _time: u32,
+    ) {
+    }
+
+    fn surface_enter(
+        &mut self,
+        _conn: &Connection,
+        _qh: &smithay_client_toolkit::reexports::client::QueueHandle<Self>,
+        _surface: &smithay_client_toolkit::reexports::client::protocol::wl_surface::WlSurface,
+        _output: &smithay_client_toolkit::reexports::client::protocol::wl_output::WlOutput,
+    ) {
+    }
+
+    fn surface_leave(
+        &mut self,
+        _conn: &Connection,
+        _qh: &smithay_client_toolkit::reexports::client::QueueHandle<Self>,
+        _surface: &smithay_client_toolkit::reexports::client::protocol::wl_surface::WlSurface,
+        _output: &smithay_client_toolkit::reexports::client::protocol::wl_output::WlOutput,
+    ) {
+    }
+}
+
+impl OutputHandler for SmithayRunnerState {
+    fn output_state(&mut self) -> &mut OutputState {
```

Comment:
We have to discuss handling multiple outputs, dynamically added new outputs and removed as well.